src/bound/mod.rs

```diff
@@ -0,0 +1,3 @@
+mod bound;
+
+pub use self::bound::*;
```

Comment:
Thanks for the review and feedback.

Yes, that makes sense 🤔. I was also not too impressed after taking another look at it this morning.

Having everything in separate directories when there is only one module makes it harder to reason about the code rather than easier.

I'll rebase this PR and update it according to your feedback :+1: